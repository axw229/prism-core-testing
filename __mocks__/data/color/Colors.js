// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
