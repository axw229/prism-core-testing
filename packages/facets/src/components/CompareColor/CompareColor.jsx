// @flow
import React from 'react'
import { fullColorNumber, getContrastYIQ } from '../../../src/shared/helpers/ColorUtils'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import 'src/providers/fontawesome/fontawesome'
import type { Color } from '../../shared/types/Colors.js.flow'
import type { FlatScene, FlatVariant } from '../../shared/types/Scene'
import WithConfigurationContext, { type ConfigurationContextType } from '../../contexts/ConfigurationContext/WithConfigurationContext'
import './CompareColor.scss'
import { FormattedMessage } from 'react-intl'
import SingleTintableSceneView from '../SingleTintableSceneView/SingleTintableSceneView'
import { createMiniColorFromColor } from '../SingleTintableSceneView/util'

const baseClass = 'color__comparison'
const containerClass = `${baseClass}__container`
const containerHeaderClass = `${baseClass}__container__header`
const containerHeaderButtonClass = `${baseClass}__container__header__button`
const wrapperClass = `${baseClass}__wrapper`
const prevBtnWrapperClass = `${baseClass}__prev-btn__wrapper`
const buttonsClass = `${baseClass}__buttons`
const buttonsVisibleClass = `${baseClass}__buttons--visible`
const nextBtnWrapperClass = `${baseClass}__next-btn__wrapper`
const backgroundColorClass = `${baseClass}__background__color`
const queueWrapperClass = `${baseClass}__queue__wrapper`

const defaultViewItem = 3
const offset = 1

type CompareColorProps = {
  colors: Color[],
  config: ConfigurationContextType,
  toggleCompareColor: Function,
  colorIds: string[],
  scenesCollection: FlatScene[],
  variantsCollection: FlatVariant[],
  selectedSceneUid: string,
  selectedVariantName: string
}

type CompareColorState = {
  colors: Color[],
  colorIds: string[],
  renderColors: Color[],
  curr: number,
  isHideNextButton: boolean,
  isHidePrevButton: boolean
}
export class CompareColor extends React.Component<CompareColorProps, CompareColorState> {
  state = {
    colors: this.props.colors,
    colorIds: this.props.colorIds,
    renderColors: [],
    curr: 0,
    isHidePrevButton: true,
    isHideNextButton: true,
    selectedSceneUid: this.props.selectedSceneUid,
    selectedVariantName: this.props.selectedVariantName
  }

  static getDerivedStateFromProps (props: CompareColorProps, state: CompareColorState) {
    if (props.colors !== state.colors) {
      return {
        colors: props.colors
      }
    }
    if (props.colorIds !== state.colorIds) {
      if (props.colorIds.length === state.colors.length - 1) {
        props.toggleCompareColor()
        return null
      }
      let currPointer = state.curr
      // when remove compared color id from color paltete
      if (props.colorIds.length > state.colorIds.length) {
        if (state.colors.length - props.colorIds.length <= defaultViewItem || currPointer === 0) {
          currPointer = 0
        } else {
          currPointer = currPointer - offset
        }
      }
      // when add compared color id from color pallate
      if (props.colorIds.length < state.colorIds.length) {
        if (state.colors.length - props.colorIds.length - 1 <= defaultViewItem) {
          currPointer = 0
        } else {
          currPointer = currPointer + 1
        }
      }
      return {
        colorIds: props.colorIds,
        curr: currPointer
      }
    }
    return null
  }

  componentDidUpdate (prevProps, prevState) {
    if (prevProps.selectedSceneUid !== this.props.selectedSceneUid ||
        prevProps.selectedVariantName !== this.props.selectedVariantName) {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({
        selectedSceneUid: this.props.selectedSceneUid,
        selectedVariantName: this.props.selectedVariantName
      })
    }
  }

  renderContent = (curr: number, colors: Colors[], colorIds: string, colorInfoClass: string, colorNumOnBottom: boolean, selectedSceneUid: string, selectedVariantName: string, scenesCollection: FlatScene[], variantsCollection: FlatVariant[]) => {
    const renderColors = colors.filter((color: Color) => {
      return !colorIds.includes(color.id)
    })
    return renderColors && renderColors.map((color, index) => {
      const selectedVariant = variantsCollection.find(variant => variant.sceneUid === selectedSceneUid && variant.variantName === selectedVariantName)
      const selectedScene = scenesCollection.find(scene => scene.uid === selectedSceneUid)
      const colorDataTemplate = selectedVariant.surfaces.map(surface => null)
      const fillFirstSurfaceColor = (template, color) => {
        const miniColor = createMiniColorFromColor(color)
        return template.map((placeholder, i) => {
          return i ? null : miniColor
        })
      }
      const colorsForView = fillFirstSurfaceColor(colorDataTemplate, color)
      return (
        <div key={index} style={{ backgroundColor: color.hex, transform: `translateX(-${curr * 100}%)` }} className={`${queueWrapperClass}`}>
          <div className={`${backgroundColorClass}`} />
          <SingleTintableSceneView
            surfaceColorsFromParents={colorsForView}
            selectedSceneUid={selectedSceneUid}
            scenesCollection={[selectedScene]}
            variantsCollection={[selectedVariant]} />
          <div style={{ color: getContrastYIQ(color.hex) }} className={`${colorInfoClass}${colorNumOnBottom ? '__name-number' : ''}`} >
            <span className={`${colorInfoClass}__number`}>{fullColorNumber(color.brandKey, color.colorNumber)}</span>
            <span className={`${colorInfoClass}__name`}>{color.name}</span>
          </div>
        </div>
      )
    })
  }

  isShowSlideButton = () => {
    const { curr, colors, colorIds } = this.state
    const itemsCount = colors.length - colorIds.length
    const buttonVisibility = {
      isHidePrevButton: true,
      isHideNextButton: true
    }
    if (curr === 0) {
      buttonVisibility.isHidePrevButton = true
    } else {
      buttonVisibility.isHidePrevButton = false
    }

    if (itemsCount > defaultViewItem && curr < itemsCount - defaultViewItem) {
      buttonVisibility.isHideNextButton = false
    }

    if (curr === itemsCount - defaultViewItem - offset / 2) {
      buttonVisibility.isHideNextButton = true
    }
    return buttonVisibility
  }

  handlePrev = () => {
    const { colors, curr, colorIds } = this.state
    const itemsCount = colors.length - colorIds.length
    if (itemsCount > defaultViewItem) {
      if (curr === (offset / 2)) {
        this.setState({ curr: curr - offset / 2 })
      } else {
        this.setState({ curr: curr - offset })
      }
    }
  }

  handleNext = () => {
    const { colors, curr, colorIds } = this.state
    const itemsCount = colors.length - colorIds.length
    if (itemsCount > defaultViewItem) {
      if (curr === itemsCount - defaultViewItem - 1) {
        this.setState({ curr: curr + offset / 2 })
      } else {
        this.setState({ curr: curr + offset })
      }
    }
  }

  render () {
    const { curr, colors, colorIds, selectedSceneUid, selectedVariantName } = this.state
    const { config, variantsCollection, scenesCollection } = this.props
    const { colorWall: { colorSwatch = {} }, cvw = {} } = config
    const { colorNumOnBottom = false, houseShaped = false } = colorSwatch
    const { closeBtn = {} } = cvw
    const { showArrow: closeBtnShowArrow = true, text: closeBtnText = <FormattedMessage id='CLOSE' /> } = closeBtn
    const colorInfoClass = houseShaped ? `${baseClass}-house-shaped` : `${baseClass}__color__info`
    const content = this.renderContent(curr, colors, colorIds, colorInfoClass, colorNumOnBottom, selectedSceneUid, selectedVariantName, scenesCollection, variantsCollection)
    const { isHidePrevButton, isHideNextButton } = this.isShowSlideButton()

    return (
        <div className={`${containerClass}`}>
          <div className={`${containerHeaderClass}`}>
            <span><FormattedMessage id='COMPARE_COLORS' /></span>
            <button className={`${containerHeaderButtonClass}`} onClick={this.props.toggleCompareColor}> {closeBtnText ?? <FormattedMessage id='CLOSE' />}{closeBtnShowArrow && <FontAwesomeIcon className={`${containerHeaderClass}--icon`} icon={['fa', 'chevron-up']} />}</button>
          </div>
          <div className={`${wrapperClass}`} >
            <div className={`${prevBtnWrapperClass}`}>
              {
                <button className={`${buttonsClass} ${isHidePrevButton ? `${buttonsVisibleClass}` : ''}`} onClick={this.handlePrev}>
                  <FontAwesomeIcon icon={['fa', 'chevron-left']} />
                </button>}
            </div>
            {content}
            <div className={`${nextBtnWrapperClass}`}>
              {
                <button className={`${buttonsClass} ${isHideNextButton ? `${buttonsVisibleClass}` : ''}`} onClick={this.handleNext}>
                  <FontAwesomeIcon icon={['fa', 'chevron-right']} />
                </button>
              }
            </div>
          </div>
        </div>
    )
  }
}

export {
  containerClass,
  containerHeaderClass,
  containerHeaderButtonClass,
  wrapperClass,
  prevBtnWrapperClass,
  buttonsClass,
  buttonsVisibleClass,
  nextBtnWrapperClass,
  backgroundColorClass,
  queueWrapperClass
}

export default WithConfigurationContext(CompareColor)
