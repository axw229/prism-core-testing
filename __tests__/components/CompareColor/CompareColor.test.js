/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import {
  CompareColor,
  containerHeaderClass,
  containerHeaderButtonClass,
  prevBtnWrapperClass,
  buttonsClass,
  nextBtnWrapperClass,
  queueWrapperClass,
  colorInfoNumberClass,
  colorInfoNameClass
} from 'src/components/CompareColor/CompareColor'
import { StaticTintScene } from 'src/components/CompareColor/TintableScene'
import * as Colors from '__mocks__/data/color/Colors'
import { fullColorNumber } from 'src/shared/helpers/ColorUtils'

const colors = Colors.getlpColorsPredefined()
const mockToggleCompareColorFn = jest.fn()
const colorsCount = colors.colors.length

const getCompareColor = props => {
  let defaultProps = {
    colors: colors.colors,
    toggleCompareColor: mockToggleCompareColorFn,
    colorsId: []
  }

  let newProps = Object.assign({}, defaultProps, props)
  return shallow(<CompareColor {...newProps} />)
}

describe('CompareColor with props', () => {
  let compareColor
  beforeEach(() => {
    if (!compareColor) {
      compareColor = getCompareColor({})
    }
  })

  it('should match snapshot with props', () => {
    expect(compareColor).toMatchSnapshot()
  })

  it('CompareColor is rendering with props', () => {
    expect(compareColor.exists()).toBe(true)
  })

  it('should initialize state correctly', () => {
    const initialState = compareColor.instance().state

    expect(initialState).toEqual({
      colors: compareColor.instance().props.colors,
      colorsId: compareColor.instance().props.colorsId,
      renderColors: [],
      curr: 0,
      isHidePrevButton: true,
      isHideNextButton: true
    })
  })

  it('should render Close button', () => {
    expect(compareColor.find(`button.${containerHeaderButtonClass}`).exists()).toBe(true)
  })

  it(`should render div.${queueWrapperClass} number of times equal to ${colorsCount}-count of colors`, () => {
    expect(compareColor.find(`div.${queueWrapperClass}`)).toHaveLength(colorsCount)
  })

  it(`should render StaticTintScene component number of times equal to ${colorsCount}-count of colors`, () => {
    expect(compareColor.find(StaticTintScene)).toHaveLength(colorsCount)
  })

  it('should render span with Compare Colors as content', () => {
    const headerContent = 'Compare Colors'
    expect(compareColor.find(`div.${containerHeaderClass} > span`).contains(headerContent)).toBe(true)
  })

  it(`should render span.${colorInfoNumberClass} number of times equal to ${colorsCount}-count of colors with content from colors prop`, () => {
    const colorNumbersArray = []
    compareColor.find(`span.${colorInfoNumberClass}`).map(el => {
      colorNumbersArray.push(el.props().children)
    })

    const colorNumbersFromPropColors = []
    compareColor.instance().props.colors.map(color => {
      colorNumbersFromPropColors.push(fullColorNumber(color.brandKey, color.colorNumber))
    })

    expect(colorNumbersArray).toEqual(colorNumbersFromPropColors)
  })

  it(`should render span.${colorInfoNameClass} number of times equal to ${colorsCount}-count of colors with content from colors prop`, () => {
    const colorNamesArray = []
    compareColor.find(`span.${colorInfoNameClass}`).map(el => {
      colorNamesArray.push(el.props().children)
    })

    const colorNamesFromPropColors = []
    compareColor.instance().props.colors.map(color => {
      colorNamesFromPropColors.push(color.name)
    })

    expect(colorNamesArray).toEqual(colorNamesFromPropColors)
  })
})

describe('CompareColor events', () => {
  let compareColor
  beforeEach(() => {
    if (!compareColor) {
      compareColor = getCompareColor({})
    }
  })

  it('should call mockToggleCompareColorFn on button.containerHeaderButtonClass click', () => {
    compareColor.find(`button.${containerHeaderButtonClass}`).simulate('click')
    expect(mockToggleCompareColorFn).toHaveBeenCalled()
  })

  it('should update curr state on next button click', () => {
    const currState = compareColor.state('curr')

    if (colorsCount > 3) {
      compareColor.find(`div.${nextBtnWrapperClass} > button.${buttonsClass}`).simulate('click')
      const currStateAfterClickNext = compareColor.state('curr')
      expect(currStateAfterClickNext).toBeGreaterThan(currState)
    }
  })

  it('should update curr state on previous button click', () => {
    compareColor.setState({ curr: 2 })
    const currState = compareColor.state('curr')

    if (colorsCount > 3) {
      compareColor.find(`div.${prevBtnWrapperClass} > button.${buttonsClass}`).simulate('click')
      const currStateAfterClickPrev = compareColor.state('curr')
      expect(currStateAfterClickPrev).toBeLessThan(currState)
    }
  })
})
