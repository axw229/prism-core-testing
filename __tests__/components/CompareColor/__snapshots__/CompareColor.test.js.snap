// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareColor with props should match snapshot with props 1`] = `
<div
  className="color__comparison__container"
>
  <div
    className="color__comparison__container__header"
  >
    <span>
      Compare Colors
    </span>
    <button
      className="color__comparison__container__header__button"
      onClick={[MockFunction]}
    >
      close Â 
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Array [
            "fa",
            "chevron-up",
          ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
  </div>
  <div
    className="color__comparison__wrapper"
  >
    <div
      className="color__comparison__prev-btn__wrapper"
    >
      <button
        className="color__comparison__buttons color__comparison__buttons--visible"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                320,
                512,
                Array [],
                "f053",
                "M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z",
              ],
              "iconName": "chevron-left",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
    <div
      className="color__comparison__queue__wrapper"
      key="0"
      style={
        Object {
          "backgroundColor": "#94766c",
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        className="color__comparison__background__color"
      />
      <StaticTintScene
        color={
          Object {
            "blue": 108,
            "brandKey": "SW",
            "brandedCollectionNames": Array [
              "Interior Historic Collection",
            ],
            "colorFamilyNames": Array [
              "Historic Color",
            ],
            "colorNumber": "0001",
            "coordinatingColors": Object {
              "coord1ColorId": "11290",
              "coord2ColorId": "11296",
            },
            "description": Array [
              "Dull",
              "Fairly Bright",
            ],
            "green": 118,
            "hex": "#94766c",
            "hue": 0.041666666666666685,
            "id": "1544",
            "isDark": false,
            "isExterior": true,
            "isInterior": true,
            "lightness": 0.5019607843137255,
            "lrv": 20.427,
            "name": "Mulberry Silk",
            "red": 148,
            "saturation": 0.15748031496062997,
            "similarColors": Array [
              "1733",
              "1719",
              "1747",
              "1754",
              "1712",
              "1775",
              "1740",
              "2831",
              "1768",
              "11269",
            ],
          }
        }
      />
      <div
        className="color__comparison__color__info"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <span
          className="color__comparison__color__info__number"
        >
          SW 0001 
        </span>
        <span
          className="color__comparison__color__info__name"
        >
          Mulberry Silk
        </span>
      </div>
    </div>
    <div
      className="color__comparison__queue__wrapper"
      key="1"
      style={
        Object {
          "backgroundColor": "#ac7559",
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        className="color__comparison__background__color"
      />
      <StaticTintScene
        color={
          Object {
            "blue": 89,
            "brandKey": "SW",
            "brandedCollectionNames": Array [
              "Color Forecast",
              "Interior Historic Collection",
            ],
            "colorFamilyNames": Array [
              "Historic Color",
            ],
            "colorNumber": "0007",
            "coordinatingColors": Object {
              "coord1ColorId": "11345",
              "coord2ColorId": "11352",
            },
            "description": Array [
              "Subdued",
              "Muted",
              "Fairly Bright",
            ],
            "green": 117,
            "hex": "#ac7559",
            "hue": 0.05622489959839356,
            "id": "1550",
            "isDark": false,
            "isExterior": true,
            "isInterior": true,
            "lightness": 0.5117647058823529,
            "lrv": 22.143,
            "name": "Decorous Amber",
            "red": 172,
            "saturation": 0.33333333333333337,
            "similarColors": Array [
              "3008",
              "3016",
              "3007",
              "1676",
              "1804",
              "2026",
              "2041",
              "2034",
              "2033",
              "3011",
            ],
          }
        }
      />
      <div
        className="color__comparison__color__info"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <span
          className="color__comparison__color__info__number"
        >
          SW 0007 
        </span>
        <span
          className="color__comparison__color__info__name"
        >
          Decorous Amber
        </span>
      </div>
    </div>
    <div
      className="color__comparison__queue__wrapper"
      key="2"
      style={
        Object {
          "backgroundColor": "#c19f6e",
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        className="color__comparison__background__color"
      />
      <StaticTintScene
        color={
          Object {
            "blue": 110,
            "brandKey": "SW",
            "brandedCollectionNames": Array [
              "Lifestyle",
              "Interior Historic Collection",
            ],
            "colorFamilyNames": Array [
              "Historic Color",
            ],
            "colorNumber": "0012",
            "coordinatingColors": Object {
              "coord1ColorId": "2881",
              "coord2ColorId": "11238",
            },
            "description": Array [
              "Fairly Colorful",
              "Moderately Colorful",
              "Fairly Bright",
            ],
            "green": 159,
            "hex": "#c19f6e",
            "hue": 0.09839357429718876,
            "id": "1555",
            "isDark": false,
            "isExterior": false,
            "isInterior": true,
            "lightness": 0.5941176470588235,
            "lrv": 37.174,
            "name": "Empire Gold",
            "red": 193,
            "saturation": 0.4009661835748792,
            "similarColors": Array [
              "3005",
              "11204",
              "11205",
              "1817",
              "2999",
              "3001",
              "1636",
              "11284",
              "2986",
              "11208",
            ],
          }
        }
      />
      <div
        className="color__comparison__color__info"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <span
          className="color__comparison__color__info__number"
        >
          SW 0012 
        </span>
        <span
          className="color__comparison__color__info__name"
        >
          Empire Gold
        </span>
      </div>
    </div>
    <div
      className="color__comparison__queue__wrapper"
      key="3"
      style={
        Object {
          "backgroundColor": "#8ca49c",
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        className="color__comparison__background__color"
      />
      <StaticTintScene
        color={
          Object {
            "blue": 156,
            "brandKey": "SW",
            "brandedCollectionNames": Array [
              "Acute Care Color Collection",
              "Interior Historic Collection",
            ],
            "colorFamilyNames": Array [
              "Historic Color",
            ],
            "colorNumber": "0017",
            "coordinatingColors": Object {
              "coord1ColorId": "1806",
              "coord2ColorId": "2926",
            },
            "description": Array [
              "Dull",
              "Fairly Bright",
            ],
            "green": 164,
            "hex": "#8ca49c",
            "hue": 0.4444444444444444,
            "id": "1560",
            "isDark": false,
            "isExterior": true,
            "isInterior": true,
            "lightness": 0.5960784313725491,
            "lrv": 34.492,
            "name": "Calico",
            "red": 140,
            "saturation": 0.11650485436893207,
            "similarColors": Array [
              "11313",
              "2158",
              "1900",
              "2165",
              "11230",
              "11222",
              "11314",
              "2144",
              "1684",
              "1574",
            ],
          }
        }
      />
      <div
        className="color__comparison__color__info"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <span
          className="color__comparison__color__info__number"
        >
          SW 0017 
        </span>
        <span
          className="color__comparison__color__info__name"
        >
          Calico
        </span>
      </div>
    </div>
    <div
      className="color__comparison__queue__wrapper"
      key="4"
      style={
        Object {
          "backgroundColor": "#75b9ae",
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        className="color__comparison__background__color"
      />
      <StaticTintScene
        color={
          Object {
            "blue": 174,
            "brandKey": "SW",
            "brandedCollectionNames": Array [
              "Lifestyle",
              "HGTV Homeâ¢ by Sherwin-Williams",
            ],
            "colorFamilyNames": Array [
              "Blue",
            ],
            "colorNumber": "6759",
            "coordinatingColors": Object {
              "coord1ColorId": "2689",
              "coord2ColorId": "11355",
              "whiteColorId": "2436",
            },
            "description": Array [
              "Subdued",
              "Muted",
              "Fairly Bright",
            ],
            "green": 185,
            "hex": "#75b9ae",
            "hue": 0.47303921568627455,
            "id": "2446",
            "isDark": false,
            "isExterior": true,
            "isInterior": true,
            "lightness": 0.592156862745098,
            "lrv": 41.426,
            "name": "Cooled Blue",
            "red": 117,
            "saturation": 0.3269230769230769,
            "similarColors": Array [
              "2439",
              "11224",
              "11223",
              "1618",
              "2447",
              "2453",
              "2173",
              "2624",
              "11232",
              "2151",
            ],
            "storeStripLocator": "163-C4",
          }
        }
      />
      <div
        className="color__comparison__color__info"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        <span
          className="color__comparison__color__info__number"
        >
          SW 6759 
        </span>
        <span
          className="color__comparison__color__info__name"
        >
          Cooled Blue
        </span>
      </div>
    </div>
    <div
      className="color__comparison__queue__wrapper"
      key="5"
      style={
        Object {
          "backgroundColor": "#d5eae8",
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        className="color__comparison__background__color"
      />
      <StaticTintScene
        color={
          Object {
            "blue": 232,
            "brandKey": "SW",
            "colorFamilyNames": Array [
              "SW6763_COLOR_FAMILY_NAME",
            ],
            "colorNumber": "6763",
            "coordinatingColors": Object {
              "coord1ColorId": "2872",
              "coord2ColorId": "1844",
            },
            "description": Array [
              "Subdued",
              "Muted",
              "Brilliant",
              "Radiant",
            ],
            "green": 234,
            "hex": "#d5eae8",
            "hue": 0.4841269841269842,
            "id": "2450",
            "isDark": false,
            "isExterior": true,
            "isInterior": true,
            "lightness": 0.8764705882352941,
            "lrv": 78.757,
            "name": "Retiring Blue",
            "red": 213,
            "saturation": 0.33333333333333304,
            "similarColors": Array [
              "2797",
              "2169",
              "2457",
              "2471",
              "2443",
              "2184",
              "2178",
              "2795",
              "2478",
              "2436",
            ],
          }
        }
      />
      <div
        className="color__comparison__color__info"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <span
          className="color__comparison__color__info__number"
        >
          SW 6763 
        </span>
        <span
          className="color__comparison__color__info__name"
        >
          Retiring Blue
        </span>
      </div>
    </div>
    <div
      className="color__comparison__queue__wrapper"
      key="6"
      style={
        Object {
          "backgroundColor": "#3aa9ae",
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        className="color__comparison__background__color"
      />
      <StaticTintScene
        color={
          Object {
            "blue": 174,
            "brandKey": "SW",
            "brandedCollectionNames": Array [
              "Kids' Colors",
            ],
            "colorFamilyNames": Array [
              "Blue",
            ],
            "colorNumber": "6767",
            "coordinatingColors": Object {
              "coord1ColorId": "2689",
              "coord2ColorId": "11359",
              "whiteColorId": "2162",
            },
            "description": Array [
              "Soft",
              "Moderately Bright",
            ],
            "green": 169,
            "hex": "#3aa9ae",
            "hue": 0.507183908045977,
            "id": "2454",
            "isDark": false,
            "isExterior": true,
            "isInterior": true,
            "lightness": 0.4549019607843137,
            "lrv": 32.458,
            "name": "Aquarium",
            "red": 58,
            "saturation": 0.5000000000000001,
            "similarColors": Array [
              "2636",
              "2632",
              "2461",
              "2626",
              "2468",
              "2447",
              "2475",
              "2174",
              "2173",
              "2474",
            ],
            "storeStripLocator": "164-C5",
          }
        }
      />
      <div
        className="color__comparison__color__info"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        <span
          className="color__comparison__color__info__number"
        >
          SW 6767 
        </span>
        <span
          className="color__comparison__color__info__name"
        >
          Aquarium
        </span>
      </div>
    </div>
    <div
      className="color__comparison__queue__wrapper"
      key="7"
      style={
        Object {
          "backgroundColor": "#c2e0e3",
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        className="color__comparison__background__color"
      />
      <StaticTintScene
        color={
          Object {
            "blue": 227,
            "brandKey": "SW",
            "brandedCollectionNames": Array [
              "Lifestyle",
            ],
            "colorFamilyNames": Array [
              "Blue",
            ],
            "colorNumber": "6771",
            "coordinatingColors": Object {
              "coord1ColorId": "11236",
              "coord2ColorId": "11284",
              "whiteColorId": "2184",
            },
            "description": Array [
              "Subdued",
              "Muted",
              "Brilliant",
              "Radiant",
            ],
            "green": 224,
            "hex": "#c2e0e3",
            "hue": 0.5151515151515151,
            "id": "2458",
            "isDark": false,
            "isExterior": true,
            "isInterior": true,
            "lightness": 0.8254901960784313,
            "lrv": 70.619,
            "name": "Bathe Blue",
            "red": 194,
            "saturation": 0.37078651685393244,
            "similarColors": Array [
              "2465",
              "2178",
              "2451",
              "2185",
              "2457",
              "2192",
              "2170",
              "2486",
              "2479",
              "2472",
            ],
            "storeStripLocator": "165-C1",
          }
        }
      />
      <div
        className="color__comparison__color__info"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <span
          className="color__comparison__color__info__number"
        >
          SW 6771 
        </span>
        <span
          className="color__comparison__color__info__name"
        >
          Bathe Blue
        </span>
      </div>
    </div>
    <div
      className="color__comparison__next-btn__wrapper"
    >
      <button
        className="color__comparison__buttons "
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                320,
                512,
                Array [],
                "f054",
                "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
              ],
              "iconName": "chevron-right",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
  </div>
</div>
`;
